app/javascript/components/concept-map/helpers/path-helpers.ts

```diff
@@ -1,82 +1,46 @@
 import {
   ConceptPath,
-  ConceptConnection,
   ConceptPathCoordinate,
-  ConceptPathState,
-} from '../concept-connection-types'
-
-import { ConceptState } from '../concept-types'
-
-import { conceptSlugToId } from '../Concept'
-
-type CategorizedConceptPaths = {
-  unavailable: ConceptPath[]
-  available: ConceptPath[]
-  completed: ConceptPath[]
+  ConceptPathStatus,
+  ConceptStatus,
+} from '../concept-map-types'
+
+import { getLineWidth, getCircleRadius } from './style-helpers'
+
+export function determinePath(
+  pathStartElement: HTMLElement,
+  pathEndElement: HTMLElement
+): ConceptPath {
+  const conceptStatus = pathEndElement.dataset.conceptStatus as ConceptStatus
+
+  return {
+    start: getPathStartFromElement(pathStartElement),
+    end: getPathEndFromElement(pathEndElement),
+    status: getPathState(conceptStatus),
+  }
 }
 
-export function determinePathTypes(
-  connections: ConceptConnection[],
-  activeConcept: string | null = null,
-  matchActive: boolean | null = null
-): CategorizedConceptPaths {
-  const paths: CategorizedConceptPaths = {
-    unavailable: [],
-    available: [],
-    completed: [],
+export function normalizePathToCanvasSize(
+  path: ConceptPath,
+  width: number,
+  height: number
+): ConceptPath {
+  const radius = getCircleRadius()
+  const lineWidth = getLineWidth()
+
+  const leftToRight = path.start.x <= path.end.x
+
+  return {
+    start: {
+      x: leftToRight ? radius + lineWidth : width - radius - lineWidth,
+      y: radius + lineWidth,
+    },
+    end: {
+      x: leftToRight ? width - radius - lineWidth : radius + lineWidth,
+      y: height - radius - lineWidth,
+    },
+    status: path.status,
```

Comment:
```suggestion
```

Reply:
This change requires the type to change which cascades to a bunch of other changes.

Perhaps the better solution is to refactor so that the generated original path **is** the normalized path.